coderd/coderd.go

```diff
@@ -298,6 +298,7 @@ func New(options *Options) *API {
 			DeploymentStats: options.AgentStatsRefreshInterval,
 		},
 	)
+	defer metricsCache.Close()
```

Comment:
Review: This is not required, but a precaution to always close services we open and not rely on code further down.